eloquentjs/11_3_real_promises/real_promises.js

```diff
@@ -0,0 +1,50 @@
+const { generateLogs } = require('./src/generateLogs');
+const { fileToArray } = require('./src/fileToArray');
+const { textFile } = require('./src/textFile');
+
+/**
+ * Returns promise of hourly security camera activity recorded for given day of week (0-6)
+ * @param {*} day 
+ * @returns 
+ */
+function activityTable(day) {
+    return new Promise((resolve, reject) => {
+        textFile('camera_logs.txt')
+        .then(file => Promise.resolve(fileToArray(file)))
```

Comment:
Is there a clearer name you could use than file? Is the thing it's storing really a file?

Also, I don't think you should be using Promise.resolve like this. I understand the intention, but you can just combine this then with the next then into a longer (but still short) little function.

Reply:
Along with the refactor of taken a stab at renaming many of the variables in this section. It feels like a hard balance between being too specific and not specific enough.

Reply:
I don't think being too descriptive is a significant problem. What are you worried about?

Reply:
It's not a significant issue for me, but I sometimes worry that being overly-descriptive would make my code cumbersome to read and write. I had briefly considered naming the `logManifest` `cameraLogRawContents` but decided against it.

Reply:
I suppose `logManifest` feels like a better description of what I intend to use that variable _for_ whereas `cameraLogRawContents` feels like a better description of what it _is_.

Reply:
"Overly-descriptive" is somewhat loaded language that I think might be coloring your thinking. "more descriptive" is a truer reflection of reality. Being more descriptive in your code makes simpler code more cumbersome to write, more complex code easier to write, and all code much easier to read. Code is generally read much more than written, so that's the problem you should optimize for.

I think a good way to think about what makes good code is that it reads like a story. Imagine you're going to explain your code to someone. You want concepts to appear in the order you'd want to cover them in, and you'd want ideas (eg variables and functions) to have the same names you'd want to refer to them as. Learning to find the words and see those patterns is quite a bit of work, but is a worthwhile challenge - it will set you apart as an engineer.

When it comes to the name you had as `logManifest`, I think using the word manifest obscures what it is too much, and I don't think cameraLogRawContents is actually what it is. I would have called it `rawCameraLogFilenames` since that's what it is.